### PR TITLE
Fix OAuth Create User event

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

This PR fixes a bug that event triggers are not being correctly fired when a new account is created with the OAuth endpoints.

Both webhooks and Function execution has been tested and confirmed to be working.

## Related PRs and Issues

#2406  #3565 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by stnguyen90** [comment]
Would it be worthwhile to wrap this logic in a function or something because there are a few places where we probably need this (creating user during creating phone session, magic url, team membership)?

File: app/controllers/api/account.php [hunk 1]
```diff
@@ -494,6 +498,61 @@
...
+                        payload: $user
+                    );
+
+                    Realtime::send(
+                        projectId: 'console',
+                        payload: $user->getArrayCopy(),
+                        events: $allEvents,
+                        channels: $target['channels'],
+                        roles: $target['roles']
+                    );
```
Comment (TorstenDittmann) [inline app/controllers/api/account.php:536]: we don't need this event for the console 👍🏻

Comment (PineappleIOnic) [reply app/controllers/api/account.php]: Resolved. 👍

File: app/controllers/api/account.php [hunk 2]
```diff
@@ -494,6 +498,53 @@
...
+                    /** Trigger Webhook */
+                    $userModel = new User();
+
+                    $userCreate = new Event(Event::WEBHOOK_QUEUE_NAME, Event::WEBHOOK_CLASS_NAME);
+                    $userCreate
+                        ->setProject($project)
+                        ->setUser($user)
+                        ->setEvent('users.[userId].create')
+                        ->setParam('userId', $userId)
+                        ->setPayload($user->getArrayCopy(array_keys(($userModel->getRules()))))
```
Comment (stnguyen90) [inline app/controllers/api/account.php:511]: What's the reasoning on this rather than just using `$user`?

**Review by stnguyen90 — COMMENTED** [review]
Deferring to @TorstenDittmann on https://github.com/appwrite/appwrite/pull/3567#issuecomment-1190644563.

File: app/controllers/api/account.php (hunk 2 above)
Comment (PineappleIOnic) [reply app/controllers/api/account.php:511]: To prevent data such as the password hash from being accidentally exposed

**Comment by karniv00l** [comment]
Sorry to push, but any progress on that? Those events are crucial for my app

**Comment by eldadfux** [comment]
@PineappleIOnic Why are replicating the entire logic for the shutdown method? I'm sure there's a way for us to add support for multiple events. We can try and change some of the terminology we use from `setEvent` to `addEvent`. Then, we can just iterate over all the available events and send them to our workers one by one.

**Review by eldadfux — CHANGES_REQUESTED** [review]
@PineappleIOnic Why are replicating the entire logic for the shutdown method? I'm sure there's a way for us to add support for multiple events. We can try and change some of the terminology we use from setEvent to addEvent. Then, we can just iterate over all the available events and send them to our workers one by one.

**Comment by stnguyen90** [comment]
@PineappleIOnic, what's the blocker on this? Looks like there are merge conflicts now.

**Comment by PineappleIOnic** [comment]
We've been trying to figure out a better way of doing this instead of manually firing the event again, preferably by updating the event middleware to support multiple events on one route. I need to find some time to sync with @TorstenDittmann to talk to him about a possible solution to this.

**Comment by stnguyen90** [comment]
@PineappleIOnic can this be closed since this isn't the approach we will be taking?

**Comment by PineappleIOnic** [comment]
Yeah, we should be able to close this. Closing it now.

**Pull request closed without merging**